python/ray/serialization.py

```diff
@@ -173,13 +207,23 @@ def add_contained_object_ref(self, object_ref):
             # a local reference that won't ever be removed.
             ray.worker.global_worker.core_worker.add_object_ref_reference(object_ref)
 
+    def restricted_loads(self, serialized_data, *, fix_imports=True,
+        encoding="ASCII", errors="strict", buffers=None):
+        if isinstance(serialized_data, str):
+            raise TypeError("Can't load pickle from unicode string")
+        file = io.BytesIO(serialized_data)
+        unpickler = RestrictedUnpickler(file, fix_imports=fix_imports, buffers=buffers,
+                      encoding=encoding, errors=errors)
+        unpickler.set_ray_pickle_whitelist(self._pickle_whitelist)
+        return unpickler.load()
+
     def _deserialize_pickle5_data(self, data):
         try:
             in_band, buffers = unpack_pickle5_buffers(data)
             if len(buffers) > 0:
-                obj = pickle.loads(in_band, buffers=buffers)
+                obj = self.restricted_loads(in_band, buffers=buffers)
             else:
-                obj = pickle.loads(in_band)
+                obj = self.restricted_loads(in_band)
```

Comment:
We can move all these code to a separate file to avoid conflicts. 
The code will be something like this.
```
def patch_pickle_for_security()
   if env_var_not_empty:
      pickle.loads = restricted_loads
```

Reply:
done.